arch/arm/src/kinetis/kinetis_flexcan.c

```diff
@@ -719,7 +719,8 @@ static int kinetis_transmit(FAR struct kinetis_driver_s *priv)
 
       uint32_t *frame_data_word = (uint32_t *)&frame->data[0];
 
-      for (int i = 0; i < (frame->len + 4 - 1) / 4; i++)
+      uint32_t i;
```

Comment:
Although code inside arch/ are not required to follow C89, please move the function declaration to the top of the function to spread good practice and avoid people follow it to core parts of the OS where it is forbidden.